ament_cmake_test/cmake/ament_add_test.cmake

```diff
@@ -86,6 +89,9 @@ function(ament_add_test testname)
     "--package-name" "${PROJECT_NAME}")
   if(ARG_SKIP_TEST)
     list(APPEND cmd_wrapper "--skip-test")
+    set(ARG_SKIP_RETURN_CODE 0)
+  elseif(ARG_SKIP_RETURN_CODE)
+    list(APPEND cmd_wrapper "--skip-return-code" "${ARG_SKIP_RETURN_CODE}")
   endif()
```

Comment:
Just to clarify this `elseif()`, unconditionally skipping a test using `SKIP_TEST` overrides the return code behavior because we'll never invoke the test and therefore will never get the magic return code to skip it because we'll ALWAYS skip it. However, since we know that we'll always skip it, setting the `SKIP_RETURN_CODE` to `0` makes CTest invoke the test so that the wrapper can write the gtest result, while still reporting the test as skipped in it's own console output and result file.